contracts/clearing/TradingApp.sol

```diff
@@ -67,21 +33,24 @@ contract TradingApp is IForceMoveApp {
 		}
 
 		bytes memory candidateData = candidate.variablePart.appData;
+
 		// settlement
-		// TODO: unsure whether we should check the proof when consensus is reached
-		if (candTurnNum % 2 == 0 && proof.length == 0) {
+		uint8 signaturesNum = NitroUtils.getClaimedSignersNum(candidate.signedBy);
+		if (candTurnNum % 2 == 0 && signaturesNum == 2 && proof.length > 0) {
 			Consensus.requireConsensus(fixedPart, proof, candidate);
```

Comment:
We should also add a check that the number of states in proof is dividable by 2.